Add What's New page intro and Releases page Licenses section (#18630)
cockroachdb/docs: src/current/_includes/releases/whats-new-intro.md

On this page, you can read about changes and find downloads for all production and testing releases of CockroachDB {{ page.major_version }}.

- For key feature enhancements in {{ page.major_version }} and other upgrade considerations, refer to the notes for {% if page.major_version != 'v1.0' %}[{{ page.major_version }}.0](#{{ page.major_version | replace: '.', '-' }}-0){% else %}{{ page.major_version }} on this page.{% endif %}

> - For details about the types of releases, their naming, and licenses, refer to the [Releases]({% link releases/index.md %}) page.

Reviewer: ```suggestion
- For details about release types, naming, and licensing, refer to the [Releases]({% link releases/index.md %}) page.
```
Otherwise, we need another "their" before "licenses. WDYT?
